proto/src/determined/project/v1/project.proto

```diff
@@ -10,42 +10,43 @@ import "google/protobuf/timestamp.proto";
 import "google/protobuf/wrappers.proto";
 import "protoc-gen-swagger/options/annotations.proto";
 
-// GeneralColumn is the pre-defined column names for experiment list table.
-enum GeneralColumn {
-  // Unspecified column.
-  GENERAL_COLUMN_UNSPECIFIED = 0;
-  // Column id.
-  GENERAL_COLUMN_ID = 1;
-  // Column name.
-  GENERAL_COLUMN_NAME = 2;
-  // Column description.
-  GENERAL_COLUMN_DESCRIPTION = 3;
-  // Column tags.
-  GENERAL_COLUMN_TAGS = 4;
-  // Column forked.
-  GENERAL_COLUMN_FORKED = 5;
-  // Column starttime.
-  GENERAL_COLUMN_STARTTIME = 6;
-  // Column duration.
-  GENERAL_COLUMN_DURATION = 7;
-  // Column trial count.
-  GENERAL_COLUMN_COUNT = 8;
-  // Column state.
-  GENERAL_COLUMN_STATE = 9;
-  // Column searcher type.
-  GENERAL_COLUMN_SEARCHER_TYPE = 10;
-  // Column resourse pool.
-  GENERAL_COLUMN_RESOURSE_POOL = 11;
-  // Column progress.
-  GENERAL_COLUMN_PROGRESS = 12;
-  // Column checkpoint size.
-  GENERAL_COLUMN_CHECKPOINT_SIZE = 13;
-  // Column checkpoint count.
-  GENERAL_COLUMN_CHECKPOINT_COUNT = 14;
-  // Column user.
-  GENERAL_COLUMN_USER = 15;
+// LocationType indicates where a column comes from
+enum LocationType {
+  // Location unknown
+  LOCATION_TYPE_UNSPECIFIED = 0;
+  // Column is located on the experiment
+  LOCATION_TYPE_EXPERIMENT = 1;
+  // Column is located in the hyperparameter config of the experiment
+  LOCATION_TYPE_HYPERPARAMETERS = 2;
+  // Column is located on the experiment's validation metrics
+  LOCATION_TYPE_VALIDATIONS = 3;
```

Comment:
we're planning to add arbitrary metric types / "generic metrics" in the future to allow users to report metric types which are user specified string , e.g., "test" or "inference".

so we have a choice here:
- either have `LOCATION_TYPE_METRICS` which encapsulates metrics of all types
- or `LOCATION_TYPE_VALIDATION`, `LOCATION_TYPE_TRAINING`, `LOCATION_TYPE_OTHER_METRICS`

the first option seems like a better fit for me because I don't think various metric "locations" will get different treatment, but I may be wrong.

Reply:
we can update the api when that lands, can't we?

Reply:
yes, although if we pick the first option, we could do it right now.

Reply:
I'm worried that this is a little speculative since you've said "but I may be wrong". If we go with what we have then a future rewrite will have updating this API as a requirement and that can inform the overall design. If we go with the suggested refactor we have a few outcomes:

1. We got exactly the correct abstraction and we save a minor update on the frontend
2. We got the abstraction wrong and have to change it anyway
3. We got the abstraction wrong but we don't change it because it technically works and it becomes tech debt

I'd rather we write some code now that we know we'll throw away than spend time speculating about the future and miss the mark.